src/reducers/account/index.js

```diff
@@ -91,7 +91,16 @@ const accessKeys = handleActions({
 
 const transactions = handleActions({
     [getTransactions]: (state, { error, payload, ready }) => {
-        const hash = state.transactions && state.transactions.reduce((h, t) => ({
+        const { transactions } = state
+
+        if (!ready && transactions && transactions[0].query_account_id !== state.accountId) {
+            return {
+                ...state,
+                transactions: undefined
```

Comment:
why `undefined` specifically vs e.g. empty array?

Reply:
At this moment if `getTransactions` is launched first time, for `!ready` we have `transactions` undefined (not empty), so I implement it the same here - thanks to that we would not have to check `transactions.length` additionally.

Reply:
should it be `if (!ready || transactions && transactions[0].query_account_id !== state.accountId)`? 

also given that we can be querying for multiple account IDs, maybe it makes sense to restructure store around this? Smth like:

```
{ transactions: { [accountId]: [ ... transaction list goes here ] } }
```

Reply:
> should it be if (!ready || transactions && transactions[0].query_account_id !== state.accountId)?

not really, we want to lunch it only for `!ready` and only if transactions list is available (transaction clear will not proceed if for example you just created a new account). The problem was that after user switch account, transactions list for the old account was visible for a short while, till the new transactions list was received (for new accountId). 

> also given that we can be querying for multiple account IDs

what do you mean?

Reply:
I mean that we can be querying different accounts, so it makes sense to organize transaction data in Redux store keyed by account. That way when you switch to different account – we can show cached data before new transaction list is received.

Reply:
Sure, so maybe we should export all transaction state to separate reducer, instead of keeping it in `account` state?

Reply:
> maybe we should export all transaction state to separate reducer, instead of keeping it in account state?

sure